app/controllers/register/company/AddCompanyDirectorsController.scala

```diff
@@ -78,4 +84,26 @@ class AddCompanyDirectorsController @Inject() (
         )
       }
   }
+
+  private def directorsWithFlag(
+                                 withFlag: Seq[(DirectorDetails, Boolean)],
+                                 directors: Seq[DirectorDetails],
+                                 flags: Seq[Boolean]
+                               ): Seq[(DirectorDetails, Boolean)] = {
+
+    if(directors.isEmpty){
+      withFlag
+    } else if (flags.isEmpty) {
+      val addWithFlag: (DirectorDetails, Boolean) = (directors.head, false)
+      directorsWithFlag(withFlag :+ addWithFlag, directors.tail, Seq.empty)
+    } else {
+      val addWithFlag: (DirectorDetails, Boolean) = (directors.head, flags.head)
```

Comment:
Could we achieve the same with foldLeft? If not, we could set this method as @tailrec then the compiler attempts to optimize it.

Reply:
@tailrec added
The reason a map or foldLeft or indeed zip, zipAll have not been used as we iterate and combine two separate collections. It should continue to iterate over directors adding a false flag as default.